src/NexusMods.App.UI/Controls/StandardButton.axaml.cs

```diff
@@ -0,0 +1,285 @@
+    {
+        get => GetValue(RightIconProperty);
+        set => SetValue(RightIconProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the label is shown on the <see cref="StandardButton"/>.
+    /// </summary>
+    public bool ShowLabel
+    {
+        get => GetValue(ShowLabelProperty);
+        set => SetValue(ShowLabelProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the type of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Types Type
+    {
+        get => GetValue(TypeProperty);
+        set => SetValue(TypeProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the size of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Sizes Size
+    {
+        get => GetValue(SizeProperty);
+        set => SetValue(SizeProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the fill option of the <see cref="StandardButton"/>.
+    /// </summary>
+    public Fills Fill
+    {
+        get => GetValue(FillProperty);
+        set => SetValue(FillProperty, value);
+    }
+
+    /// <inheritdoc/>
+    protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+    {
+        base.OnApplyTemplate(e);
+
+        _leftIcon = e.NameScope.Find<UnifiedIcon>("PART_LeftIcon");
+        _rightIcon = e.NameScope.Find<UnifiedIcon>("PART_RightIcon");
+        _label = e.NameScope.Find<TextBlock>("PART_Label");
+        _content = e.NameScope.Find<ContentPresenter>("PART_ContentPresenter");
+        _border = e.NameScope.Find<Border>("PART_Border");
+
+        if (_leftIcon == null || _rightIcon == null || _label == null || _content == null || _border == null) return;
+
+        _leftIcon.Value = LeftIcon;
+        _rightIcon.Value = RightIcon;
+
+        _label.IsVisible = ShowLabel;
+
+        // so we can use the traditional button as well as our own properties to set the button
+        if (Content is null)
+        {
+            _content.IsVisible = false;
+            _border.IsVisible = true;
+        }
+        else
+        {
+            _content.IsVisible = true;
+            _border.IsVisible = false;
+        }
+
+        switch (ShowIcon)
+        {
+            case ShowIconOptions.None:
+                _leftIcon!.IsVisible = false;
+                _rightIcon!.IsVisible = false;
+                break;
+            case ShowIconOptions.Left:
+                _leftIcon!.IsVisible = true;
+                _rightIcon!.IsVisible = false;
+                break;
+            case ShowIconOptions.Right:
+                _leftIcon!.IsVisible = false;
+                _rightIcon!.IsVisible = true;
+                break;
+            case ShowIconOptions.Both:
+                _leftIcon!.IsVisible = true;
+                _rightIcon!.IsVisible = true;
+                break;
+            default:
+                _leftIcon!.IsVisible = false;
+                _rightIcon!.IsVisible = false;
+                break;
+        }
```

Comment:
This code can be simplified a bit like this I think:
The idea is that you set the value directly rather than doing it in conditional branches.

```suggestion
        _content.IsVisible = Content is not null;
        _border.IsVisible = Content is null;
        
        _leftIcon.IsVisible = ShowIcon is ShowIconOptions.Left or ShowIconOptions.Both;
        _rightIcon.IsVisible = ShowIcon is ShowIconOptions.Right or ShowIconOptions.Both;
```